misc-scripts/gifts_import/populate_protein_feature.pl

```diff
@@ -0,0 +1,350 @@
+  my $giftsDB = {
+    user => $gfuser,
+    pass => $gfpass,
+    host => $gfhost,
+    port => $gfport,
+    dbname => $gfdb
+
+  };
+
+# This is where the gifts info will be written to
+  my $protfeatureDB = {
+    user   => $pfuser,
+    pass   => $pfpass,
+    host   => $pfhost,
+    port   => $pfport,
+    dbname => $pfdb
+  };
+
+  return ( $giftsDB, $protfeatureDB );
+}
+
+
+populate_protein_feature_db();
+
+sub get_analysis_id {
+  my ($logic_name) = @_;
+  my $analysis_id;
+  
+  my $dba = get_dba($protfeatureDB);
+  my $analysis_dba = $dba->get_AnalysisAdaptor();
+  my $analysis = $analysis_dba->fetch_by_logic_name($logic_name);
+  
+  $analysis_id = $analysis->dbID() if defined $analysis;
+  return $analysis_id;
+}
+
+
+sub populate_protein_feature_db {
+
+  my $gifts_dbh = db_connect($giftsDB );
+  my $protfeature_dbh = db_connect($protfeatureDB);
+  
+  if($gifts_dbh){
+    print "Got connection to GiftsDB \n";
+  }
+  
+  if($protfeature_dbh){
+    print "Got connection to ProtfeatureDB \n";
+  }
+  
+  my $insert_container = build_insert_sql($gifts_dbh, $protfeature_dbh);
+
+  
+  my $rows_inserted = 0;
+  my @failed_stmts = [];
+  unless($dryrun){
+
+  foreach my $insert_stmt (@$insert_container) {
+    eval{
+    my $affected = $protfeature_dbh->do($insert_stmt);
+    $rows_inserted += $affected;
+    };
+    if($@){
+      print "$@\n";
+      push(@failed_stmts, $@);
+    }
+    
+  }
+  print Dumper(\@failed_stmts);
+  print "INSERT END\n";
+  }
+
+  print "Number of rows inserted into protein_feature $protfeatureDB->{'dbname'}.protein_feature ", $rows_inserted, "\n";
+  return $rows_inserted;
+  
+  $gifts_dbh->disconnect();
+  $protfeature_dbh->disconnect();
+  
+}
+
+sub annotate_ref{
+  my $ref = shift;
+  my $ensp_id = $ref->[0];
+  my $ensp = $translation_adaptor->fetch_by_stable_id( $ensp_id ) ;
+  if($ensp){
+  $ref->[0] = $ensp->dbID;
+  $ref->[1] = 1;
+  $ref->[2] = $ensp->length()-1;
+  
+  }
+  
+  return $ref;
+}
+
+sub build_insert_sql {
+  my ($gifts_dbh, $protfeature_dbh) = @_;
+  
+  #get analysis id
+  my $analysis_id = get_analysis_id("gifts_import");
+  print("Analysis id $analysis_id \n");
+  
+  my $select_sql = qq{SELECT DISTINCT ensp_id, 0, 0, uniprot_acc, $analysis_id as analysis_id, mdz, 'mdtag' FROM ensp_u_cigar};
+  $select_sql = $limit ? $select_sql . " limit 10" : $select_sql;
+  
+  my $insert_sql = qq{INSERT INTO $protfeatureDB->{'dbname'}.protein_feature(translation_id, seq_start, seq_end, hit_name, analysis_id, cigar_line, align_type) VALUES };
+
+
+  my $import_sth = $gifts_dbh->prepare($select_sql);
+  $import_sth->execute;
+  my $rows = $import_sth->rows;
+
+  my $start = 0;
+  my $insert_values;
+  my $max_rows_in_insert = 9999;    #Batch size of 10,000
+  # my $max_rows_in_insert = 10;    #Batch size of 10
+  my $rows_in_insert     = 0;
+
+  my @insert_container = ();
+  my $miss = 1;
+  while ( my $ref = $import_sth->fetchrow_arrayref ) {
+     my $not_ensp =1;
```

Comment:
Not that it seems to be used later, but why is this starting at 1 if it's an incrementer for non-matching lines?